haskell/test/Lorentz/Contracts/Test/Permit.hs

```diff
@@ -347,18 +348,49 @@ permitSpec originate = do
           withOriginated originate defaultOriginationParams $ \stablecoinContract -> do
             withSender testPauser $ do
               hash <- callPermit stablecoinContract testPauserPK testPauserSK 0 Pause
-              lCallEP stablecoinContract (Call @"Set_expiry") $ SetExpiryParam 5 (Just (hash, testPauser))
-              lCallEP stablecoinContract (Call @"Set_expiry") $ SetExpiryParam 3 (Just (hash, testPauser))
+              lCallEP stablecoinContract (Call @"Set_expiry") $ SetExpiryParam testPauser 5 $ Just hash
+              lCallEP stablecoinContract (Call @"Set_expiry") $ SetExpiryParam testPauser 3 $ Just hash
             withSender wallet1 $ do
               rewindTime 4
               lCallEP stablecoinContract (Call @"Pause") () `catchExpectedError` errExpiredPermit
 
+      it "can only be accessed by the issuer of the permit" $
+        integrationalTestExpectation $ do
+          withOriginated originate defaultOriginationParams $ \stablecoinContract -> do
+            withSender wallet1 $ do
+              pausePermitHash <- callPermit stablecoinContract testPauserPK testPauserSK 0 Pause
+              lCallEP stablecoinContract
+                (Call @"Set_expiry") (SetExpiryParam testPauser 10 $ Just pausePermitHash)
+                `catchExpectedError` errNotPermitIssuer
+
+      it "can be accessed via a permit" $
+        integrationalTestExpectation $ do
+          withOriginated originate defaultOriginationParams $ \stablecoinContract -> do
+            withSender wallet1 $ do
+              pausePermitHash <- callPermit stablecoinContract testPauserPK testPauserSK 0 Pause
+              callPermit stablecoinContract testPauserPK testPauserSK 1 $
+                Set_expiry $ SetExpiryParam testPauser 1000 $ Just pausePermitHash
+              lCallEP stablecoinContract (Call @"Set_expiry") $
+                SetExpiryParam testPauser 1000  $ Just pausePermitHash
+              lCallEP stablecoinContract (Call @"Pause") ()
+
+      it "should revoke the permit when set to zero" $
+        integrationalTestExpectation $ do
+          withOriginated originate defaultOriginationParams $ \stablecoinContract -> do
+            withSender wallet1 $ do
+              pausePermitHash <- callPermit stablecoinContract testPauserPK testPauserSK 0 Pause
+              callPermit stablecoinContract testPauserPK testPauserSK 1 $
+                Set_expiry $ SetExpiryParam testPauser 0 $ Just pausePermitHash
+              lCallEP stablecoinContract (Call @"Set_expiry") $
+                SetExpiryParam testPauser 0  $ Just pausePermitHash
+              lCallEP stablecoinContract (Call @"Pause") () `catchExpectedError` mgmNotPauser
```

Comment:
If I understand this correctly, I believe this can be simplified by using `withSender testPauser` and deleting the call to `callPermit ... Set_expiry`

```suggestion
            withSender testPauser $ do
              pausePermitHash <- callPermit stablecoinContract testPauserPK testPauserSK 0 Pause
              lCallEP stablecoinContract (Call @"Set_expiry") $
                SetExpiryParam testPauser 0  $ Just pausePermitHash
            lCallEP stablecoinContract (Call @"Pause") () `catchExpectedError` mgmNotPauser
```

Reply:
@dcastro But the `testPauser` can call the `pause` entrypoint without requiring a permit, right?

Reply:
That's right. In my suggested change, the call to `pause` is sent from `nettest`, not from `testPauser`. Maybe it's not easy to spot here on github, but the call to `pause` is not inside the `withSender testPauser` block.

So it should still attempt to find a permit, and fail because we've set expiry=0

Reply:
Right.  Changed.